### ISSUE TITLE
Loss becomes nan after some epochs in colab

### ISSUE THREAD

**Issue by Arvi63** [pythonlessons/TensorFlow-2.x-YOLOv3#8]
`epoch:49 step:    0/19, lr:0.000052, giou_loss:   2.64, conf_loss:  11.12, prob_loss:   0.08, total_loss:  13.83
epoch:49 step:    1/19, lr:0.000052, giou_loss:   2.31, conf_loss:  10.61, prob_loss:   0.08, total_loss:  12.99

giou_val_loss:    nan, conf_val_loss:  12.31, prob_val_loss:   0.06, total_val_loss:    nan

epoch:50 step:    2/19, lr:0.000052, giou_loss:   1.96, conf_loss:  10.34, prob_loss:   0.09, total_loss:  12.38
epoch:50 step:    3/19, lr:0.000052, giou_loss:   2.51, conf_loss:  11.29, prob_loss:   0.08, total_loss:  13.87
epoch:50 step:    4/19, lr:0.000052, giou_loss:   2.17, conf_loss:  10.95, prob_loss:   0.06, total_loss:  13.18`

**Comment by pythonlessons** [comment]
Try to reduce starting learning rate

**Comment by Arvi63** [comment]
I have reduced the learning rate but still loss becomes nan after some epochs.
Do you have any suggestion or solutions?

**Comment by Arvi63** [comment]
After adjusting the learning rate ,model trained without nan but it gets stuck at  total_loss: 5    after increasing the epoch number as well. Have any idea to decrease the loss further more??

**Comment by pythonlessons** [comment]
try to reduce even more, or there is something wrong with the dataset, so that model can't learn from it

**Comment by Tenesh** [comment]
I had the similar issue, i tried reducing the epoch number and did trained again by using checkpoints multiple times.

**Comment by Deep-learning999** [comment]
418/452, lr:0.000007, giou_loss:   2.78, conf_loss:  17.23, prob_loss:   0.13, total_loss:  20.15
epoch: 7 step:  419/452, lr:0.000007, giou_loss:   5.54, conf_loss:  18.36, prob_loss:   0.24, total_loss:  24.14
epoch: 7 step:  420/452, lr:0.000007, giou_loss:   2.87, conf_loss:  17.82, prob_loss:   0.10, total_loss:  20.79
epoch: 7 step:  421/452, lr:0.000007, giou_loss:   2.17, conf_loss:  16.20, prob_loss:   0.12, total_loss:  18.48
epoch: 7 step:  422/452, lr:0.000007, giou_loss:   4.43, conf_loss:  16.48, prob_loss:   0.16, total_loss:  21.07

**Comment by Deep-learning999** [comment]
total loss can't reduce

**Comment by pythonlessons** [comment]
Everything depends on your training data and evaluation data

**Comment by Myron1996** [comment]
Can someone @pythonlessons  please explain what these parameters mean with respect to training & testing data ? 
![Capture](https://user-images.githubusercontent.com/72124771/96089028-29a22780-0ebe-11eb-8ecc-255827f96774.JPG)

**Loss**
- loss/conf_loss
tag: loss/conf_loss

- loss/giou_loss
tag: loss/giou_loss

- loss/prob_loss
tag: loss/prob_loss

- loss/total_loss
tag: loss/total_loss

- Ir

**Validate loss**
- validate_loss/conf_val

- validate_loss/giou_val

- validate_loss/prob_val

- validate_loss/total_val
